larevt/CalibrationDBI/Providers/DBDataset.cxx

```diff
@@ -0,0 +1,254 @@
+
+  clear();
+  fNRows = getNtuples(dataset) - kNUMBER_HEADER_ROWS;
+  //std::cout << "DBDataset: Number of rows = " << fNRows << std::endl;
+  fChannels.reserve(fNRows);
+
+  // Process header rows.
+
+  int err = 0;
+  char buf[kBUFFER_SIZE];
+  Tuple tup;
+
+  // Extract IOV begin time.
+
+  tup = getTuple(dataset, 0);
+  getStringValue(tup, 0, buf, kBUFFER_SIZE, &err);
+  fBeginTime = IOVTimeStamp::GetFromString(std::string(buf));
+  //std::cout << "DBDataset: Begin time stamp input = " << buf << std::endl;
+  //std::cout << "DBDataset: Begin time stamp = " << fBeginTime.DBStamp() << std::endl;
+  releaseTuple(tup);
+
+  // Extract IOV end time.
+  
+  tup = getTuple(dataset, 1);
+  getStringValue(tup, 0, buf, kBUFFER_SIZE, &err);
+  if ( 0 == strcmp(buf,"-") )
+    fEndTime = IOVTimeStamp::MaxTimeStamp();
+  else
+    fEndTime = IOVTimeStamp::GetFromString(std::string(buf));
+  //std::cout << "DBDataset: End time stamp input = " << buf << std::endl;
+  //std::cout << "DBDataset: End time stamp = " << fEndTime.DBStamp() << std::endl;
+  releaseTuple(tup);
+
+  // Extract column names.
+
+  tup = getTuple(dataset, 2);
+  fNCols = getNfields(tup);
+  //std::cout << "DBDataset: Number of columns = " << fNCols << std::endl;
+  fColNames.reserve(fNCols);
+  for (size_t col=0; col < fNCols; ++col) {
+    getStringValue(tup, col, buf, kBUFFER_SIZE, &err);
+    //std::cout << "DBDataset: Column name = " << buf << std::endl;
+    fColNames.push_back(std::string(buf));
+  }
+  releaseTuple(tup);
+
+  // Extract column types.
+
+  tup = getTuple(dataset, 3);
+  fColTypes.reserve(fNCols);
+  for (size_t col=0; col < fNCols; ++col) {
+    getStringValue(tup, col, buf, kBUFFER_SIZE, &err);
+    //std:: cout << "DBDataset: Column type = " << buf << std::endl;
+    fColTypes.push_back(std::string(buf));
+  }
+  releaseTuple(tup);
+
+  // Extract data.  Loop over rows.
+
+  fData.reserve(fNRows * fNCols);
+  for(size_t row = 0; row < fNRows; ++row) {
+    //std::cout << "\nRow " << row << std::endl;
+    tup = getTuple(dataset, row + kNUMBER_HEADER_ROWS);
+
+    // Loop over columns.
+
+    for(size_t col = 0; col < fNCols; ++col) {
+      getStringValue(tup, col, buf, kBUFFER_SIZE, &err);
+
+      // Convert string value to DBDataset::value_type (std::variant).
+
+      if(fColTypes[col] == "integer" || fColTypes[col] == "bigint") {
+	long value = strtol(buf, 0, 10);
+	fData.push_back(value_type(value));
+	//std::cout << "DBDataset: row=" << row << ", column=" << col << ", value=" << fData.back() 
+	//	<< std::endl;
+	if(col == 0) {
+	  fChannels.push_back(value);
+	  //std::cout << "DBDataset: channel=" << fChannels.back() << std::endl;
+	}
+      }
+      else if(fColTypes[col] == "real") {
+	double value = strtod(buf, 0);
+	fData.push_back(value_type(value));
+	//std::cout << "DBDataset: row=" << row << ", column=" << col << ", value=" << fData.back() 
+	//	<< std::endl;
+	if(col == 0) {
+	  std::cout << "First column has wrong type real." << std::endl;
```

Comment:
Suggest removing all `std::cout` statements.  If you absolutely require printout to the terminal, then the message-facility utilities should be used.  Otherwise, the exception message already includes the error message.